src/pyOpenMS/pyopenms/dataframes.py

```diff
@@ -143,13 +143,31 @@ def get_df(self):
 class FeatureMapDF(FeatureMap):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+    def __get_prot_id_filename_from_pep_id(self, pep_id):
+        """Gets the primary MS run path of the ProteinIdentification linked with the given PeptideIdentification.
+
+        Parameters:
+        pep_id: PeptideIdentification
+
+        Returns:
+        str: primary MS run path (filename) of the ProteinIdentification with the same identifier as the given PeptideIdentification
+        """
+        for prot in self.getProteinIdentifications():
+            if prot.getIdentifier() == pep_id.getIdentifier():
+                filenames = []
+                prot.getPrimaryMSRunPath(filenames)
+                if filenames:
+                    return filenames[0]
+        return 'unknown'
     
     # meta_values = None (default), 'all' or list of meta value names
-    def get_df(self, meta_values = None):
+    def get_df(self, meta_values = None, export_peptide_identifications = True):
         """Generates a pandas DataFrame with information contained in the FeatureMap.
 
         Parameters:
         meta_values: meta values to include (None, [custom list of meta value names] or 'all')
+        export_peptide_identifications (bool): export best hit for PeptideIdentifications assigned to a feature
```

Comment:
do we need to add a note here how all peptide hits can be retrieved and linked back to the feature